src/arch/x86/tlb.cc

```diff
@@ -94,8 +94,14 @@ TLB::evictLRU()
 }
 
 TlbEntry *
-TLB::insert(Addr vpn, const TlbEntry &entry)
+TLB::insert(Addr vpn, const TlbEntry &entry, uint64_t pid)
```

Comment:
I don't think we want to use pid generally. I think we want to use an ASID/PCID. On Intel machines, the PCID is the lower 12 bits of the CR3. We could just use that.

https://xem.github.io/minix86/manual/intel-x86-and-64-manual-vol3/o_fe12b1e2a880e0ce-139.html

Same thing for AMD. See section 5.1.1
https://www.amd.com/system/files/TechDocs/24593.pdf

Reply:
Updated this change to use PCID rather than pid.